### PR TITLE
Doc/improve mermaid chart

### LINKED ISSUE #27: Consider a Mermaid diagram?
Currently the "architecture diagram" is a PNG file https://github.com/open-telemetry/opentelemetry-demo-webstore/blob/b28cb38af2f1f1273f92a321d713e0f7706536b6/docs/img/architecture-diagram.png.

I imagine we'll keep improving the architecture, and PNG is hard for change management / comment. I propose that we use mermaid. In OpenTelemetry .NET and C++ I've seen good results https://github.com/open-telemetry/opentelemetry-dotnet/issues/2952.

If folks like the idea, I can take this work.

### PR DESCRIPTION
**Pull request by julianocosta89**
Fixes #27.

## Changes

Added grpc & http communication labels to mermaid chart.

### CODE REVIEW

**Comment by julianocosta89** [comment]
A preview of the chart can be seen in here:
https://github.com/julianocosta89/opentelemetry-demo/tree/doc/improve-mermaid-chart#architecture

**Review by reyang — APPROVED** [review]

**Comment by julianocosta89** [comment]
Checks / markdownlinkcheck failure not related to the PR:
```
ERROR: 2 dead links found!
[✖] https://github.com/austinlparker → Status: 500
[✖] https://github.com/mtwo → Status: 500
```

Any ideas how to solve this?

**Comment by reyang** [comment]
> Checks / markdownlinkcheck failure not related to the PR:
> 
> ```
> ERROR: 2 dead links found!
> [✖] https://github.com/austinlparker → Status: 500
> [✖] https://github.com/mtwo → Status: 500
> ```
> 
> Any ideas how to solve this?

Wait for GitHub to recover and rerun the CI. Or ask the maintainer to forcibly merge the PR with the understanding that these failures were caused by GitHub transilient issue rather than caused by this PR.

**Review by mic-max — APPROVED** [review]

**Pull request merged**
